### PR TITLE
Fix to pull docker images from docker hub if not found locally or remotely

### LINKED ISSUE #422: [Bug] Docker hub images are not pulled
**Describe the bug**
When an image is run that is not local and not on harbor, it is not pulled from docker hub.

```python
2022-12-06 09:33:17 - task_manager   - WARNING  - Could not construct remote URL, are you using a local image?
2022-12-06 09:33:17 - task_manager   - WARNING  - Or an image from docker hub?
2022-12-06 09:33:17 - task_manager   - WARNING  - We'll make an final attempt when running the image to pull it without any checks...
2022-12-06 09:33:17 - task_manager   - ERROR    - Cannot locate image itisfoundation/sleeper
2022-12-06 09:33:18 - node           - ERROR    - 404 Client Error: Not Found ("no such image: itisfoundation/sleeper: No such image: itisfoundation/sleeper:latest")
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/docker/api/client.py", line 261, in _raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http+docker://localhost/v1.35/images/itisfoundation/sleeper/json

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/vantage6/vantage6-node/vantage6/node/__init__.py", line 792, in run_forever
    self.__start_task(task)
  File "/vantage6/vantage6-node/vantage6/node/__init__.py", line 351, in __start_task
    database=task.get('database', 'default')
  File "/vantage6/vantage6-node/vantage6/node/docker/docker_manager.py", line 301, in run
    database=database
  File "/vantage6/vantage6-node/vantage6/node/docker/task_manager.py", line 205, in run
    vpn_ports = self._run_algorithm()
  File "/vantage6/vantage6-node/vantage6/node/docker/task_manager.py", line 258, in _run_algorithm
    algo_image_name=self.image
  File "/vantage6/vantage6-node/vantage6/node/docker/vpn_manager.py", line 226, in forward_vpn_traffic
    helper_container, algo_image_name)
  File "/vantage6/vantage6-node/vantage6/node/docker/vpn_manager.py", line 289, in _forward_traffic_to_algorithm
    ports = self._find_exposed_ports(algo_image_name)
  File "/vantage6/vantage6-node/vantage6/node/docker/vpn_manager.py", line 340, in _find_exposed_ports
    n2n_image = self.docker.images.get(image)
  File "/usr/local/lib/python3.7/site-packages/docker/models/images.py", line 316, in get
    return self.prepare_model(self.client.api.inspect_image(name))
  File "/usr/local/lib/python3.7/site-packages/docker/utils/decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/docker/api/image.py", line 246, in inspect_image
    self._get(self._url("/images/{0}/json", image)), True
  File "/usr/local/lib/python3.7/site-packages/docker/api/client.py", line 267, in _result
    self._raise_for_status(response)
  File "/usr/local/lib/python3.7/site-packages/docker/api/client.py", line 263, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/usr/local/lib/python3.7/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.ImageNotFound: 404 Client Error: Not Found ("no such image: itisfoundation/sleeper: No such image: itisfoundation/sleeper:latest")
```

**To Reproduce**
1. run task using a docker image (which is not local)
2. see node logs

I can fix it by pulling the image manually

### PR DESCRIPTION
**Pull request by bartvanb**
Fix #422

### CODE REVIEW

**Comment by frankcorneliusmartin** [comment]
I am not sure if this is the solution. The command:

` client.container.run('hello-world')` pulls the image when it is not found locally. We should investigate why that is currently not the case in the infrastructure. 

The `pull_if_newer` method should only be used with an harbor instance.

File: vantage6-common/vantage6/common/docker/addons.py [hunk 1]
```diff
@@ -238,9 +238,9 @@ def pull_if_newer(docker_client, image: str, log=ClickLogger):
             log.warn(f"Local image is newer! Are you testing? {image}")
```
Comment (frankcorneliusmartin) [inline vantage6-common/vantage6/common/docker/addons.py:238]: `pull = False`

Comment (frankcorneliusmartin) [reply vantage6-common/vantage6/common/docker/addons.py:238]: And remove the `pull=True` from the `remote_ > local_` case.

Comment (frankcorneliusmartin) [reply vantage6-common/vantage6/common/docker/addons.py:238]: add `pull=False` to the `remote_ == local_`

File: vantage6-common/vantage6/common/docker/addons.py [hunk 2]
```diff
@@ -238,9 +238,9 @@ def pull_if_newer(docker_client, image: str, log=ClickLogger):
...
     elif local_:
         log.warn(f"Only a local image has been found! {image}")
+        pull = False
     elif remote_:
         log.debug("No local image found, pulling from remote!")
-        pull = True
     elif not local_ and not remote_:
         log.error(f"Cannot locate image {image}")
```
Comment (frankcorneliusmartin) [inline vantage6-common/vantage6/common/docker/addons.py:245]: Maybe tell them we didn't find it on an Harbor instance?

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]

File: vantage6-common/vantage6/common/docker/addons.py (hunk 1 above)
Comment (bartvanb) [reply vantage6-common/vantage6/common/docker/addons.py]: instead, changed it back and added extra line that was missing before

**Review by frankcorneliusmartin — APPROVED** [review]
lgtm

**Pull request merged**